tests/tests_unit/test_config.py

```diff
@@ -108,14 +108,25 @@ def test_load(self, credentials):
 
     @pytest.mark.parametrize("protocol", ("http", "https"))
     @pytest.mark.parametrize("end", ("", "/", ":8080", "/api/v1/", ":8080/api/v1/"))
-    def test_extract_cdf_cluster(self, client_config, protocol, end):
-        for valid in ("3D", "my_clus-ter", "jazz-testing-asia-northeast1-1", "trial-00ed82e12d9cbadfe28e4"):
-            client_config.base_url = f"{protocol}://{valid}.cognitedata.com{end}"
-            assert client_config.cdf_cluster == valid
+    @pytest.mark.parametrize("subdomain", ("", "p001.plink."))
+    @pytest.mark.parametrize(
+        "cluster", ("3D", "my_clus-ter", "jazz-testing-asia-northeast1-1", "trial-00ed82e12d9cbadfe28e4")
+    )
+    def test_extract_valid_cdf_cluster(
+        self, client_config: ClientConfig, protocol: str, end: str, subdomain: str, cluster: str
+    ) -> None:
+        client_config.base_url = f"{protocol}://{subdomain}{cluster}.cognitedata.com{end}"
+        assert client_config.cdf_cluster == cluster
 
-        for invalid in ("", ".", "..", "huh.my_cluster."):
-            client_config.base_url = f"{protocol}://{valid}cognitedata.com{end}"
-            assert client_config.cdf_cluster is None
+    @pytest.mark.parametrize("protocol", ("http", "https"))
+    @pytest.mark.parametrize("end", ("", "/", ":8080", "/api/v1/", ":8080/api/v1/"))
+    @pytest.mark.parametrize("cluster", ("", ".", "..", "huh.my_cluster."))
+    def test_extract_invalid_cdf_cluster(
+        self, client_config: ClientConfig, protocol: str, end: str, cluster: str
+    ) -> None:
+        client_config.base_url = f"{protocol}://{cluster}cognitedata.com{end}"
+        assert client_config.cdf_cluster is None
```

Comment:
The "negative" tests is missing subdomain. Thanks for fixing the bug in the existing test code btw, I spotted it while reviewing 😅 
```py
# invalid is unused... :facepalm:
for invalid in ("", ".", "..", "huh.my_cluster."):
    client_config.base_url = f"{protocol}://{valid}cognitedata.com{end}"
    assert client_config.cdf_cluster is None
```